File: src/MP.Server/ApiServer.Model/Model/ViewModel/SysUser.cs
using System;

namespace ApiServer.Model.Model.ViewModel
{
    public class SysUser
    {
        public long id { get; set; }

        /// <summary>
        /// 用户名
        /// </summary>
        public string username { get; set; }

        /// <summary>
        /// 密码
        /// </summary>
        public string password { get; set; }

        /// <summary>
        /// 昵称
        /// </summary>
        public string nickname { get; set; }

        /// <summary>
        /// 头像图片路径
        /// </summary>
        public string portrait { get; set; }


        /// <summary>
        /// 组织id
        /// </summary>
        public long orgId { get; set; }

        /// <summary>
        /// 0无效用户，1是有效用户
        /// </summary>
        public bool? enabled { get; set; }

        /// <summary>
        /// 手机号
        /// </summary>
        public string phone { get; set; }

        /// <summary>
        /// email
        /// </summary>
        public string email { get; set; }

        /// <summary>
        /// 用户创建时间
        /// </summary>
        public DateTime createTime { get; set; }
    }
}

File: src/MP.Admin/src/api/system/sys_user.js
import  { jwtServerInstance }  from '../index'
import {getTokenUser} from "@/lib/utils";
import qs from	'qs'

//JWT登录认证接口
export const login = (userName,passWord) => {
  return jwtServerInstance.request({
    url:'/JwtAuth/authentication',
    method:'post',
    data:{
      username: userName,
      password: passWord
    }
  })
}
//JWT令牌刷新接口
export const refreshToken = () => {
  return jwtServerInstance.request({
    url:'/JwtAuth/refreshtoken',
    method:'get'
  })
}

//获取用户信息接口(个人中心)
export const getUserInfo = () => {
  return jwtServerInstance.request({
    url:'/sysuser/info',
    method:'get',
    params:{
      username: getTokenUser()
    }
  })
}


//=====================用户管理接口开始========================
//用户列表查询接口
export const getUsers = (queryform,pagination) => {
  return jwtServerInstance.request({
    url:'/sysuser/query',
    method:'post',
    data:qs.stringify({
      orgId: queryform.orgId,
      username: queryform.username,
      phone: queryform.phone,
      email: queryform.email,
      enabled: queryform.enabled,
      createStartTime: queryform.timeRange[0],
      createEndTime: queryform.timeRange[1],
      pageNum: pagination.pageNum,
      pageSize: pagination.pageSize
    })
  })
}

//更新sys_user的一条数据记录
export const updateUser = (userForm) => {
  return jwtServerInstance.request({
    url:'/sysuser/update',
    method:'post',
    data:userForm
  })
}

//新增一条sys_user数据记录
export const addUser = (userForm) => {
  return jwtServerInstance.request({
    url:'/sysuser/add',
    method:'post',
    data:userForm
  })
}

//删除一条sys_user数据记录
export const deleteUser = (userId) => {
  return jwtServerInstance.request({
    url:'/sysuser/delete',
    method:'post',
    data:qs.stringify({
      userId:userId
    })
  })
}

//sys_user数据记录的enabled字段
export const changeEnabled = (userId,enabled) => {
  return jwtServerInstance.request({
    url:'/sysuser/enabled/change',
    method:'post',
    data:qs.stringify({
      userId,
      enabled
    })
  })
}

//重置用户密码
export const resetUserPwd = (userId) => {
  return jwtServerInstance.request({
    url:'/sysuser/pwd/reset',
    method:'post',
    data:qs.stringify({
      userId:userId
    })
  })
}

//判断当前用户是否仍在使用默认密码，没做修改
export const pwdIsDefault = () => {
  return jwtServerInstance.request({
    url:'/sysuser/pwd/isdefault',
    method:'post',
    data:qs.stringify({
      username:getTokenUser()
    })
  })
}


//修改密码
export const changeUserPwd = (oldPass,newPass) => {
  return jwtServerInstance.request({
    url:'/sysuser/pwd/change',
    method:'post',
    data:qs.stringify({
      username:getTokenUser(),
      oldPass:oldPass,
      newPass:newPass
    })
  })
}



